### PR TITLE
corrected syntax error for counting digits in py and added new approach to check for Natural Square

### PR DESCRIPTION
**Pull request by rahul1809**
**Changes:**
[ Corrected the syntax issue in /cosmos/code/mathematical-algorithms/count_digits/count_digits.py ]
[ Added a separate file for check for natural squares using binary search ]

### CODE REVIEW

File: code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#include <bits/stdc++.h>
```
Comment (arnavb) [inline code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp]: Don't include this. See #1915.

File: code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		return true;
+
+	}
+	return false;
+}
+
+int main() {
+
+	int t;
+	cout << "Enter the Number of testcases:" << '\n';
```
Comment (arnavb) [inline code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp]: Number should not be capitalized. (Just nitpicking here).

File: code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	cin >> t;
+	lld number;
+	while (t--)
+	{
+		cin >> number;
+		bool result = IsSquare(number);
+		if (result)
+		{
+			cout << "Natural Square number" << '\n';
```
Comment (arnavb) [inline code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp]: I suggest that in thes two blocks, the removal of the braces would improve readability.

File: code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+int main() {
+
+	int t;
+	cout << "Enter the Number of testcases:" << '\n';
+	cin >> t;
+	lld number;
+	while (t--)
+	{
+		cin >> number;
+		bool result = IsSquare(number);
```
Comment (arnavb) [inline code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp]: You don't really need this variable.

File: code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+typedef long long int lld;
+bool IsSquare(lld number)
+{
+	lld min = 1;
+	lld max = number;
+	lld mid = min + (max - min)/2;
+	if (number == 0 || number == 1)
+		return true;
+	while (min<max) {
+	if (mid*mid > number)
```
Comment (arnavb) [inline code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp]: Fix your indentation inside the while loop.

File: code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp (hunk 4 above)
Comment (rahul1809) [reply code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp]: removed the variable

File: code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp (hunk 3 above)
Comment (rahul1809) [reply code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp]: removed them.

File: code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp (hunk 2 above)
Comment (rahul1809) [reply code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp]: did the change

File: code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp (hunk 5 above)
Comment (rahul1809) [reply code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp]: fixed indentation

File: code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp (hunk 1 above)
Comment (rahul1809) [reply code/mathematical-algorithms/Check_is_square/FindingISquare_binary_search.cpp]: remove and replaced by iostream header file

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**